### PR TITLE
Refactor of VM instance pool management

### PR DESCRIPTION
**Pull request by marko-gacesa**

### CODE REVIEW

File: command/delegate/delegate.go [hunk 1]
```diff
@@ -424,13 +403,10 @@ func (c *delegateCommand) handleDestroy(poolManager *vmpool.Manager) http.Handle
...
 		fmt.Printf("\n\nExecuting cleanup: %v\n", reqData)
 
-		pool := poolManager.Get(reqData.Pool)
-		instance := &vmpool.Instance{
-			ID: reqData.ID,
-			IP: "", // TODO remove this
-		}
-		destroyErr := pool.Destroy(r.Context(), instance)
-		if destroyErr != nil {
+		ctx := r.Context()
```
Comment (tphoney) [inline command/delegate/delegate.go:406]: nit no need for this line.

File: internal/vmpool/pool.go [hunk 2]
```diff
@@ -2,13 +2,19 @@ package vmpool
 
 import (
 	"context"
+	"sync"
 )
 
+const RunnerName = "drone-runner-aws"
```
Comment (tphoney) [inline internal/vmpool/pool.go:8]: the runner should not be a constant, it is the name of the machine that starts the runner https://github.com/drone-runners/drone-runner-aws/blob/4a3a405b58f4c1293275d2ecc8b41f926e67cba0/command/delegate/delegate.go#L60

**Review by tphoney — CHANGES_REQUESTED** [review]

File: command/delegate/delegate.go (hunk 1 above)
Comment (marko-gacesa) [reply command/delegate/delegate.go:406]: how's that? the `ctx` is used just below.

File: internal/vmpool/pool.go (hunk 2 above)
Comment (marko-gacesa) [reply internal/vmpool/pool.go:8]: This is not a runner name. This is actually used for the "creator" tag. It must be fixed.

File: command/delegate/delegate.go (hunk 1 above)
Comment (tphoney) [reply command/delegate/delegate.go:406]: why instantiate a variable, when you can just call the method on line 408

**Review by tphoney — APPROVED** [review]

**Pull request merged**